### PR TITLE
Feature/m2 1696 add subscale lookup table

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    try {
+      setData(JSON.parse(tableData));
+    } catch {
+      console.log('Error while table data parsing.');
+    }
+  }, [tableData]);
+
+  useEffect(() => {
+    if (tableData) return setModalType(ModalType.Edit);
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:38]: setModalType(tableData ? ModalType.Edit : ModalType.Upload)

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { useSubscaleLookupTableSetup } from './SubscaleLookupTable.hooks';
+
+export const SubscaleLookupTable = ({
+  open,
+  subscaleName,
+  tableData,
+  onUpdate,
+  onClose,
+}: SubscaleLookupTableProps) => {
+  const { modalType, step, data, error, setModalType, setError, setStep, onFileReady } =
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx:15]: pls use  const props =  useSubscaleLookupTableSetup({ tableData });
 const screens = getComponent(props)

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const [step, setStep] = useState<Steps>(0);
+  const [data, setData] = useState<DataTableItem[]>();
+  const [error, setError] = useState<JSX.Element | null>(null);
+
+  const onFileReady = (file: ImportedFile | null) => {
+    const mappedData = (file?.data ?? []).map((item) => ({
+      ...item,
+      id: uuidv4(),
+    }));
+    setData(mappedData);
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:19]: Do y need to set setData([]) in some cases?

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  };
+
+  useEffect(() => {
+    if (!tableData?.length) return;
+
+    try {
+      setData(JSON.parse(tableData));
+    } catch {
+      console.log('Error while table data parsing.');
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:33]: mb better to use console.warn or error

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.utils.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  data,
+  error,
+  onFileReady,
+  onUpdate,
+  onClose,
+  setModalType,
+  setStep,
+  setError,
+}: {
+  modalType: ModalType;
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.utils.tsx:118]: better to move this to types

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.const.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+const { t } = i18n;
+
+export const columnData = [
+  {
+    key: 'Score',
+    label: t('Score'),
+  },
+  {
+    key: 'Raw Score',
+    label: t('Raw Score'),
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.const.tsx:12]: It seems here has to be t('rawScore') and in other labels key need to be with small first letter, for example t('score')

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx (hunk 2 above)
Comment (alex-i-e) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx:15]: why? propreties from `useSubscaleLookupTableSetup({ tableData })` distributes attributes not only for `getComponent`. Moreover, there are some properties from SubscaleLookupTable need to add for `getComponent`

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts (hunk 3 above)
Comment (alex-i-e) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:19]: in some cases it's possible, it depends on provided data from csv file. What is your idea?

Comment (kdanilenka32) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:23]: const mappedData = file?.data?.map((item) =>

setData(mappedData ?? [ ]);

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+export const SubscaleLookupTable = ({
+  open,
+  subscaleName,
+  tableData,
+  onUpdate,
+  onClose,
+}: SubscaleLookupTableProps) => {
+  const {
+    modalType,
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx:16]: pls use const props  = useSubscaleLookupTableSetup({ tableData });

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    setError,
+    setStep,
+    onFileReady,
+    onDownloadTemplate,
+  } = useSubscaleLookupTableSetup({ tableData });
+
+  if (modalType === null) return null;
+
+  const screens = getComponents({
+    modalType,
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.tsx:30]: getComponents(props)

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    }));
+    setData(mappedData);
+    setError(null);
+
+    if (file) {
+      setStep((prevState) => ++prevState as Steps);
+    }
+  };
+
+  const onDownloadTemplate = async () => {
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts:31]: no need async here

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**